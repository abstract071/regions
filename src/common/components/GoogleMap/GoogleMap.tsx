import React from 'react'
import {
  GoogleMap,
  Marker,
  useLoadScript
} from '@react-google-maps/api'
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'
import CircularProgress from '@material-ui/core/CircularProgress'

import classes from './GoogleMap.module.scss'


interface IMapProps {
  latitude: number
  longitude: number
}

const Map: React.FC<IMapProps> = ( {
  latitude,
  longitude
} ) => {
  const {
    isLoaded,
    loadError
  } = useLoadScript( {
    // Sensitive data, should be used on backend or as environment variable
    googleMapsApiKey: 'AIzaSyCEkw4NOJRaKZT_kFcg0LezAmjD_NprPMc'
  } )

  if ( !latitude || !longitude ) {
    return (
      <Typography component="h6" variant="h6">
        There are no coordinates
      </Typography>
    )
  }

  if ( loadError ) {
    return <Box>Map cannot be loaded right now, sorry.</Box>
  }

  return isLoaded ? (
    <GoogleMap
      mapContainerClassName={ classes['map'] }
      center={ {
        lat: latitude,
        lng: longitude
      } }
      zoom={ 3 }
    >
      <Marker
        position={ {
          lat: latitude,
          lng: longitude
        } }
      />
    </GoogleMap>
  ) : (
    <CircularProgress />
  )
}

export default Map
