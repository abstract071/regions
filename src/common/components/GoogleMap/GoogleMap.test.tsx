import React from 'react'
import { shallow } from 'enzyme'

import GoogleMap from './GoogleMap'


describe( 'GoogleMap', () => {
  let shallowedComponent: any = null

  beforeEach( () => {
    shallowedComponent = shallow(
      <GoogleMap
        latitude={ 0 }
        longitude={ 0 }
      />
    )
  } )

  it( 'should match snapshot', () => {
    expect( shallowedComponent ).toMatchSnapshot()
  } )
} )
