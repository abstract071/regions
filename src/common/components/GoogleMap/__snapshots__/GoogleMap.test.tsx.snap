// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap should match snapshot 1`] = `ShallowWrapper {}`;
